sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/MetricStorage.java

```diff
@@ -25,5 +27,9 @@
    * @param epochNanos The timestamp for this collection.
    * @return The {@link MetricData} from this collection period, or {@code null}.
    */
-  MetricData collectAndReset(long startEpochNanos, long epochNanos);
+  MetricData collectAndReset(
+      CollectionHandle collector,
+      Set<CollectionHandle> allCollectors,
```

Comment:
Two new params could use a bit of javadoc (yeah, it's internal...but I'm trying to figure it all out, so it would probably be helpful)

Reply:
Definitely! This is why I asked for early review, will update javadocs to make this more clear.

Reply:
Added javadooc